pulp_container/app/registry_api.py

```diff
@@ -389,15 +388,37 @@ def create_dr(self, path, request):
         return distribution, repository
 
 
-def _get_pull_through_remote_data(root_cache_distribution):
-    remote_data = models.ContainerPullThroughRemote.objects.filter(
-        pk=root_cache_distribution.remote_id
-    ).values()[0]
+def _get_pull_through_remote_data(remote_data):
     for attr in IGNORED_PULL_THROUGH_REMOTE_ATTRIBUTES:
         remote_data.pop(attr, None)
     return remote_data
 
 
+def _filter_remotes(path):
+    """
+    Filter the remote repositories based on includes and excludes fields definitions
+    """
+    pull_through_cache_distribution = (
+        models.ContainerPullThroughDistribution.objects.annotate(path=Value(path))
+        .filter(path__startswith=F("base_path"))
+        .order_by("-base_path")
+        .first()
+    )
+    if not pull_through_cache_distribution:
+        raise RepositoryNotFound(name=path)
+
+    upstream_name = path.split(pull_through_cache_distribution.base_path, maxsplit=1)[1].strip("/")
+    pull_through_remote = models.ContainerPullThroughRemote.objects.filter(
+        pk=pull_through_cache_distribution.remote_id
+    ).values()[0]
+    if not filter_resource(
+        upstream_name, pull_through_remote.get("includes"), pull_through_remote.get("excludes")
+    ):
+        raise RepositoryNotFound(name=path)
```

Comment:
I wonder if it would not be better to return an HTTP 403 error instead (https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/403). What do you think?

Reply:
hum... when I see a 403 I think about roles/rights and, in this new filter feature, we are not denying or blocking users access, it feels more like "not making it visible" or just removing/bypassing a resource from the results as a convenience and not denying it since they could create a new pull-through remote with different includes/excludes ... and maybe a 403 could "misdirect" into a role troubleshoot. Do you agree?

Reply:
Yes, you are right. Let's leave it unchanged then.